datasets/metashift/metashift.py

```diff
@@ -0,0 +1,398 @@
+# Copyright 2022 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# _preprocess_groups(), _parse_node_str(), _load_candidate_subsets() adapted from here :
+# https://github.com/Weixin-Liang/MetaShift/blob/main/dataset/generate_full_MetaShift.py
+
+# MIT License
+
+# Copyright (c) 2021 Weixin-Liang
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""MetaShift Dataset."""
+
+import json
+import os
+import pickle
+from collections import Counter, defaultdict
+
+import datasets
+from datasets.tasks import ImageClassification
+
+
+_CITATION = """\
+@InProceedings{liang2022metashift,
+title={MetaShift: A Dataset of Datasets for Evaluating Contextual Distribution Shifts and Training Conflicts},
+author={Weixin Liang and James Zou},
+booktitle={International Conference on Learning Representations},
+year={2022},
+url={https://openreview.net/forum?id=MTex8qKavoS}
+}
+"""
+
+
+_DESCRIPTION = """\
+The MetaShift is a dataset of datasets for evaluating distribution shifts and training conflicts.
+The MetaShift dataset is a collection of 12,868 sets of natural images across 410 classes.
+It was created for understanding the performance of a machine learning model across diverse data distributions.
+"""
+
+_HOMEPAGE = "https://metashift.readthedocs.io/"
+
+
+_LICENSE = "https://github.com/Weixin-Liang/MetaShift/blob/main/LICENSE"
```

Comment:
here it must be the license of the data

```suggestion
_LICENSE = "Creative Commons Attribution 4.0 International License"
```